main/internal/implementations/dragon_drop/http_dragondrop_managed_visualization_test.go

```diff
@@ -0,0 +1,853 @@
+
+	// Then
+	require.NoError(t, err)
+	require.Equal(t, `{"hashicorp/google":{"4.77.0":1}}`, terraformFootprint.VersionsTFProviders)
+	require.Equal(t, `{"1.5.1":1}`, terraformFootprint.VersionsTF)
+	require.Equal(t, `{"./app-cluster":{"0.0.5":1,"0.0.6":1}}`, terraformFootprint.VersionsTFModules)
+}
+
+func TestHTTPDragonDropClient_getTerraformFootprintData_another_tf_file_multiple_modules_multiple_files(t *testing.T) {
+	// Given
+	defer cleanMockedDirectory()
+
+	ctx := context.Background()
+	client := HTTPDragonDropClient{
+		config: HTTPDragonDropClientConfig{
+			WorkspaceDirectories: []string{"workspace1"},
+		},
+	}
+
+	err := writeFile("another.tf", []byte(`
+		terraform {
+		  required_version = "1.5.1"
+		
+		  required_providers {
+			google = {
+			  source  = "hashicorp/google"
+			  version = "~>4.77.0"
+			}
+		  }
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file1.tf", []byte(`
+		module "servers" {
+			source = "./app-cluster"
+			version = "0.0.5"
+		
+			servers = 5
+		}
+
+		module "servers2" {
+			source = "./app-cluster"
+			version = "0.0.6"
+		
+			servers = 5
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file2.tf", []byte(`
+		module "server4" {
+			source = "./app-cluster"
+			version = "0.0.7"
+		
+			servers = 5
+		}
+
+		module "servers3" {
+			source = "./app-cluster"
+			version = "0.0.8"
+		
+			servers = 5
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	// When
+	terraformFootprint, err := client.getTerraformFootprintData(ctx)
+
+	// Then
+	require.NoError(t, err)
+	require.Equal(t, `{"hashicorp/google":{"4.77.0":1}}`, terraformFootprint.VersionsTFProviders)
+	require.Equal(t, `{"1.5.1":1}`, terraformFootprint.VersionsTF)
+	require.Equal(t, `{"./app-cluster":{"0.0.5":1,"0.0.6":1,"0.0.7":1,"0.0.8":1}}`, terraformFootprint.VersionsTFModules)
+}
+
+func TestHTTPDragonDropClient_getTerraformFootprintData_another_tf_file_sum_module_versions(t *testing.T) {
+	// Given
+	defer cleanMockedDirectory()
+
+	ctx := context.Background()
+	client := HTTPDragonDropClient{
+		config: HTTPDragonDropClientConfig{
+			WorkspaceDirectories: []string{"workspace1"},
+		},
+	}
+
+	err := writeFile("another.tf", []byte(`
+		terraform {
+		  required_version = "1.5.1"
+		
+		  required_providers {
+			google = {
+			  source  = "hashicorp/google"
+			  version = "~>4.77.0"
+			}
+		  }
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file1.tf", []byte(`
+		module "servers" {
+			source = "./app-cluster"
+			version = "0.0.5"
+		
+			servers = 5
+		}
+
+		module "servers2" {
+			source = "./app-cluster"
+			version = "0.0.6"
+		
+			servers = 5
+		}
+	`), "repo/workspace1")
+	require.NoError(t, err)
+
+	err = writeFile("module_file2.tf", []byte(`
+		module "server4" {
+			source = "./app-cluster"
+			version = "0.0.5"
+		
+			servers = 5
+		}
+
+		module "servers3" {
+			source = "./app-cluster"
```

Comment:
```suggestion
			source = "./app-cluster-type-2"
```
Let's make sure this edge case is handled as well (multiple module types)